### DISCUSSION TITLE
Rule Persistence in emqx under Kubernetes

### DISCUSSION THREAD

**Discussion by ctellechea2001** [emqx/emqx#5719] [Q&A]
Hi, how are you? Could you please, help me with a support? I need to know what another methods can we use to EQMX Rule Persistent in Kubernetes. (Currently we are using PVC, to provide the data persistence). But PVC is attached to a EBS volume in a specific AZ. We need to increase the high availability, for this reason we are looking another option.  

Thanks in advance!

**Comment by k32** [comment]
Hello,

From the application perspective, EMQX stores the rules in Mnesia database. It's an embedded distributed database that has a persistent replica of the data on every EMQ X node. When a new node joins the cluster, it becomes another replica of the database. So as long as at least one EMQ X node is alive, the rules won't be lost. 
I am not an expert in Kubernetes, so I cannot give you a precise recommendation, but as long as you have an up and running EMQ X instance in every region, you have a rule replica in this region.
I hope this helps.
